sdk/eventhub/azure-eventhubs/azure/eventhub/_producer.py

```diff
@@ -151,29 +154,24 @@ def _on_outcome(self, outcome, condition):
         self._outcome = outcome
         self._condition = condition
 
-    def create_batch(self, max_size=None, partition_key=None):
-        # type:(int, str) -> EventDataBatch
-        """
-        Create an EventDataBatch object with max size being max_size.
-        The max_size should be no greater than the max allowed message size defined by the service side.
-
-        :param max_size: The maximum size of bytes data that an EventDataBatch object can hold.
-        :type max_size: int
-        :param partition_key: With the given partition_key, event data will land to
-         a particular partition of the Event Hub decided by the service.
-        :type partition_key: str
-        :return: an EventDataBatch instance
-        :rtype: ~azure.eventhub.EventDataBatch
-        """
-
-        if not self._max_message_size_on_link:
-            self._open_with_retry()
-
-        if max_size and max_size > self._max_message_size_on_link:
-            raise ValueError('Max message size: {} is too large, acceptable max batch size is: {} bytes.'
-                             .format(max_size, self._max_message_size_on_link))
-
-        return EventDataBatch(max_size=(max_size or self._max_message_size_on_link), partition_key=partition_key)
+    def _wrap_eventdata(self, event_data, span, partition_key):
+        if isinstance(event_data, EventData):
+            if partition_key:
+                set_message_partition_key(event_data.message, partition_key)
+            wrapper_event_data = event_data
+            trace_message(wrapper_event_data.message, span)
+        else:
+            if isinstance(event_data, EventDataBatch):  # The partition_key in the param will be omitted.
+                if partition_key and partition_key != event_data._partition_key:  # pylint: disable=protected-access
+                    raise EventDataError('The partition_key does not match the one of the EventDataBatch')
```

Comment:
would it be overkill to log this?  (If it'll get logged within close proximity at an upper level, that'd be fine too)

Reply:
You think a user is likely to hit this in prod?
This seems like the kind of error one would hit during development and once fixed would be unlikely to show up again - unless you think some could fall into this at runtime?

Reply:
Not necessarily, but after my recent experience debugging the reconnect symptoms, I've started erring on the side of "more verbosity of needed."  Given your statements, maybe a _very_ cautious level (e.g. debug) is merited, but you know the flows better than I do so I'll let you be the judge to my "would it be overkill" statement.